azure-sdk-build-tool/src/main/java/com/microsoft/azure/sdk/build/tool/util/MavenUtils.java

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.azure.sdk.build.tool.util;
+
+import com.microsoft.azure.sdk.build.tool.util.logging.Logger;
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.model.Dependency;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+/**
+ * Utility class to perform Maven related operations.
+ */
+public class MavenUtils {
+    private static final Logger LOGGER = Logger.getInstance();
+
+    /**
+     * Creates artifact string representation of an artifact.
+     * @param artifact The artifact.
+     * @return The string representation of the artifact.
+     */
+    public static String toGAV(Artifact artifact) {
+        return artifact.getGroupId() + ":" + artifact.getArtifactId() + ":" + artifact.getVersion();
+    }
+
+    /**
+     * Creates artifact string representation of a dependency.
+     * @param dependency The dependency.
+     * @return The string representation of the dependency.
+     */
+    public static String toGAV(Dependency dependency) {
+        return dependency.getGroupId() + ":" + dependency.getArtifactId() + ":" + dependency.getVersion();
+    }
+
+    /**
+     * Gets the latest released version of the given artifact from Maven repository.
+     * @param groupId The group id of the artifact.
+     * @param artifactId The artifact id of the artifact.
+     * @return The latest version or {@code null} if an error occurred while retrieving the latest
+     * version.
+     */
+    public static String getLatestArtifactVersion(String groupId, String artifactId) {
```

Comment:
this method can be possibly simplified if it's acceptable to introduce another dependency `jackson-dataformat-xml`.

Reply:
It's an additional dependency that may not be required for this simple usecase.